### ISSUE TITLE
Epoll ADD fails on bad file descriptor

### ISSUE THREAD

**Issue by PeterGottesman** [open-mpi/ompi#5520]
On MTT we are seeing this issue pretty consistently on the v2.x branch, usually in tests where the exit code is 53. It seems that the issue frequently, but not exclusively, occurs with MPI_THREAD_MULTIPLE set.

```
[warn] Epoll ADD(4) on fd 55 failed.  Old events were 0; read change was 0 (none); write change was
1 (add): Bad file descriptor
[warn] Epoll ADD(4) on fd 87 failed.  Old events were 0; read change was 0 (none); write change was
1 (add): Bad file descriptor
[warn] Epoll ADD(4) on fd 79 failed.  Old events were 0; read change was 0 (none); write change was
1 (add): Bad file descriptor
...
```
MTT Link: https://mtt.open-mpi.org/index.php?do_redir=2697

I'm working on getting it reproduced now, but being that 2.1.4 is up for release soon I figured the sooner this is posted the better.

**Comment by rhc54** [comment]
Indicates event lib not cleanly shutting down

**Comment by jsquyres** [comment]
This is almost certainly not going to get fixed in the v2.1.x series, since v4.0.x is immanent (which then obsoletes the v2.1.x series).
